22-Mar-22 10:46:05 - INFO - Start of validation on files for training
22-Mar-22 10:46:05 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
22-Mar-22 10:46:05 - INFO - Bad_Raw directory deleted before starting validation.
22-Mar-22 10:46:05 - INFO - GoodRaw directory deleted successfully.
22-Mar-22 10:46:21 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
22-Mar-22 10:46:21 - INFO - Column length validation has been started.
22-Mar-22 10:46:21 - INFO - Column length validation has been completed.
22-Mar-22 10:46:21 - INFO - Missing values validation has been started.
22-Mar-22 10:46:21 - INFO - Raw data validation is completed.
22-Mar-22 10:46:26 - INFO - Creating training database and tables on the basis of given schema
22-Mar-22 10:46:54 - INFO - Opened websites keyspace successfully.
22-Mar-22 10:46:54 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
22-Mar-22 10:46:55 - INFO - Closed websites keyspace successfully.
22-Mar-22 10:46:55 - INFO - Table creation is completed.
22-Mar-22 10:46:55 - INFO - Insertion of data into the database is started.
22-Mar-22 10:47:03 - INFO - Opened websites keyspace successfully.
22-Mar-22 10:49:56 - INFO - The file has been inserted: no need to reinsert.
22-Mar-22 10:49:56 - INFO - Insertion into the database is completed.
22-Mar-22 10:49:56 - INFO - Deleting good data folder
22-Mar-22 10:49:56 - INFO - GoodRaw directory deleted successfully.
22-Mar-22 10:49:56 - INFO - Good_Data folder is deleted
22-Mar-22 10:49:56 - INFO - Moving bad files to archive and deleting Bad_Data folder
22-Mar-22 10:51:29 - INFO - Bad files moved to archive
22-Mar-22 10:51:29 - INFO - Bad Raw Data Folder Deleted successfully.
22-Mar-22 10:51:29 - INFO - Bad files are moved to archive. Bad folder is deleted.
22-Mar-22 10:51:29 - INFO - Validation operation is completed.
22-Mar-22 10:51:29 - INFO - Extracting csv file from the database
22-Mar-22 10:51:37 - INFO - Opened websites keyspace successfully.
22-Mar-22 10:53:35 - INFO - The csv file is exported successfully.
22-Mar-22 10:53:35 - INFO - Start of model training
22-Mar-22 10:53:35 - ERROR - Unsuccessful end of training
22-Mar-22 11:00:00 - INFO - Start of validation on files for training
22-Mar-22 11:00:00 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
22-Mar-22 11:00:01 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
22-Mar-22 11:00:01 - INFO - Column length validation has been started.
22-Mar-22 11:00:01 - INFO - Column length validation has been completed.
22-Mar-22 11:00:01 - INFO - Missing values validation has been started.
22-Mar-22 11:00:01 - INFO - Raw data validation is completed.
22-Mar-22 11:00:01 - INFO - Creating training database and tables on the basis of given schema
22-Mar-22 11:00:11 - INFO - Opened websites keyspace successfully.
22-Mar-22 11:00:11 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
22-Mar-22 11:00:12 - INFO - Closed websites keyspace successfully.
22-Mar-22 11:00:12 - INFO - Table creation is completed.
22-Mar-22 11:00:12 - INFO - Insertion of data into the database is started.
22-Mar-22 11:00:19 - INFO - Opened websites keyspace successfully.
22-Mar-22 11:01:55 - INFO - The file has been inserted: no need to reinsert.
22-Mar-22 11:01:55 - INFO - Insertion into the database is completed.
22-Mar-22 11:01:55 - INFO - Deleting good data folder
22-Mar-22 11:01:55 - INFO - GoodRaw directory deleted successfully.
22-Mar-22 11:01:55 - INFO - Good_Data folder is deleted
22-Mar-22 11:01:55 - INFO - Moving bad files to archive and deleting Bad_Data folder
22-Mar-22 11:01:55 - INFO - Bad files moved to archive
22-Mar-22 11:01:55 - INFO - Bad Raw Data Folder Deleted successfully.
22-Mar-22 11:01:55 - INFO - Bad files are moved to archive. Bad folder is deleted.
22-Mar-22 11:01:55 - INFO - Validation operation is completed.
22-Mar-22 11:01:55 - INFO - Extracting csv file from the database
22-Mar-22 11:02:05 - INFO - Opened websites keyspace successfully.
22-Mar-22 11:03:45 - INFO - The csv file is exported successfully.
22-Mar-22 11:03:45 - INFO - Start of model training
22-Mar-22 11:03:45 - INFO - Entered the get_data method of the DataGetter class
22-Mar-22 11:03:46 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
22-Mar-22 11:03:46 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
22-Mar-22 11:03:48 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
22-Mar-22 11:03:48 - INFO - Entered the is_null_present method of the Preprocessor class
22-Mar-22 11:03:48 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
22-Mar-22 11:03:48 - INFO - Entered the separate_label_feature method of the Preprocessor class
22-Mar-22 11:03:48 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
22-Mar-22 11:03:48 - INFO - Entered the elbow_plot method of the KMeansClustering class
22-Mar-22 11:04:37 - INFO - The optimum number of clusters is 2.             Exited the elbow_plot method of the KMeansClustering class
22-Mar-22 11:04:37 - INFO - Entered the create_clusters method of the KMeansClustering class
22-Mar-22 11:04:38 - INFO - Entered the save_model method of the FileOperation class
22-Mar-22 11:04:38 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
22-Mar-22 11:04:38 - INFO - Successfully created 2 clusters.             Exited the create_clusters method of the KMeansClustering class
22-Mar-22 11:04:39 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
22-Mar-22 11:04:39 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
22-Mar-22 11:04:39 - INFO - Entered the standardize_data method of the Preprocessor class
22-Mar-22 11:04:40 - INFO - The data has been standardized
22-Mar-22 11:04:41 - INFO - Entered the get_best_model method of the ModelFinder class
22-Mar-22 11:04:41 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
22-Mar-22 11:04:53 - ERROR - Exception occured in get_best_params_for_xgboost method of the ModelFinder class.             Exception message: [11:04:53] D:\bld\xgboost-split_1645118015404\work\src\data\data.cc:981: Check failed: valid: Input data contains `inf` or `nan`
22-Mar-22 11:04:53 - INFO - XGBoost parameter tuning failed.             Exited the get_best_params_for_xgboost method of the Model_Finder class
22-Mar-22 11:04:53 - ERROR - Exception occured in get_best_model method of the ModelFinder class.             Exception message: 
22-Mar-22 11:04:53 - INFO - Model selection Failed. Exited the get_best_model method of the Model_Finder class
22-Mar-22 11:04:53 - ERROR - Unsuccessful end of training
22-Mar-22 11:14:35 - INFO - Start of validation on files for training
22-Mar-22 11:14:35 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
22-Mar-22 11:14:43 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
22-Mar-22 11:14:43 - INFO - Column length validation has been started.
22-Mar-22 11:14:43 - INFO - Column length validation has been completed.
22-Mar-22 11:14:43 - INFO - Missing values validation has been started.
22-Mar-22 11:14:43 - INFO - Raw data validation is completed.
22-Mar-22 11:14:43 - INFO - Creating training database and tables on the basis of given schema
22-Mar-22 11:15:06 - INFO - Opened websites keyspace successfully.
22-Mar-22 11:15:06 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
22-Mar-22 11:15:06 - INFO - Closed websites keyspace successfully.
22-Mar-22 11:15:06 - INFO - Table creation is completed.
22-Mar-22 11:15:06 - INFO - Insertion of data into the database is started.
22-Mar-22 11:15:13 - INFO - Opened websites keyspace successfully.
22-Mar-22 11:16:57 - INFO - The file has been inserted: no need to reinsert.
22-Mar-22 11:16:57 - INFO - Insertion into the database is completed.
22-Mar-22 11:16:57 - INFO - Deleting good data folder
22-Mar-22 11:16:57 - INFO - GoodRaw directory deleted successfully.
22-Mar-22 11:16:57 - INFO - Good_Data folder is deleted
22-Mar-22 11:16:57 - INFO - Moving bad files to archive and deleting Bad_Data folder
22-Mar-22 11:16:57 - INFO - Bad files moved to archive
22-Mar-22 11:16:57 - INFO - Bad Raw Data Folder Deleted successfully.
22-Mar-22 11:16:57 - INFO - Bad files are moved to archive. Bad folder is deleted.
22-Mar-22 11:16:57 - INFO - Validation operation is completed.
22-Mar-22 11:16:57 - INFO - Extracting csv file from the database
22-Mar-22 11:18:01 - INFO - Opened websites keyspace successfully.
22-Mar-22 11:20:12 - INFO - The csv file is exported successfully.
22-Mar-22 11:20:12 - INFO - Start of model training
22-Mar-22 11:20:12 - INFO - Entered the get_data method of the DataGetter class
22-Mar-22 11:20:13 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
22-Mar-22 11:20:13 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
22-Mar-22 11:20:14 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
22-Mar-22 11:20:14 - INFO - Entered the is_null_present method of the Preprocessor class
22-Mar-22 11:20:14 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
22-Mar-22 11:20:14 - INFO - Entered the separate_label_feature method of the Preprocessor class
22-Mar-22 11:20:14 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
22-Mar-22 11:20:14 - INFO - Entered the elbow_plot method of the KMeansClustering class
22-Mar-22 11:20:50 - INFO - The optimum number of clusters is 2.             Exited the elbow_plot method of the KMeansClustering class
22-Mar-22 11:20:50 - INFO - Entered the create_clusters method of the KMeansClustering class
22-Mar-22 11:20:51 - INFO - Entered the save_model method of the FileOperation class
22-Mar-22 11:20:51 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
22-Mar-22 11:20:51 - INFO - Successfully created 2 clusters.             Exited the create_clusters method of the KMeansClustering class
22-Mar-22 11:20:52 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
22-Mar-22 11:20:52 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
22-Mar-22 11:20:52 - INFO - Entered the standardize_data method of the Preprocessor class
22-Mar-22 11:20:53 - INFO - The data has been standardized
22-Mar-22 11:20:53 - INFO - Entered the get_best_model method of the ModelFinder class
22-Mar-22 11:20:53 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
22-Mar-22 11:50:17 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 9, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
22-Mar-22 11:50:19 - INFO - ROC-AUC for XGBoost: 0.9720303882839526
22-Mar-22 11:50:19 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
22-Mar-22 21:30:07 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
22-Mar-22 21:31:16 - INFO - ROC-AUC for SVM: {self.svm_score}
22-Mar-22 21:31:17 - INFO - Entered the save_model method of the FileOperation class
22-Mar-22 21:31:21 - INFO - Model File XGBoost0 saved. Exited the save_model method of the FileOperation class
22-Mar-22 21:31:29 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
22-Mar-22 21:31:39 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
22-Mar-22 21:31:39 - INFO - Entered the standardize_data method of the Preprocessor class
22-Mar-22 21:31:40 - INFO - The data has been standardized
22-Mar-22 21:31:40 - INFO - Entered the get_best_model method of the ModelFinder class
22-Mar-22 21:31:40 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
22-Mar-22 21:32:21 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 9, 'n_estimators': 100}             Exited the get_best_params_for_xgboost method of the ModelFinder class
22-Mar-22 21:32:21 - INFO - ROC-AUC for XGBoost: 0.9719259746909518
22-Mar-22 21:32:21 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
22-Mar-22 21:32:41 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
22-Mar-22 21:32:41 - INFO - ROC-AUC for SVM: {self.svm_score}
22-Mar-22 21:32:41 - INFO - Entered the save_model method of the FileOperation class
22-Mar-22 21:32:41 - INFO - Model File XGBoost1 saved. Exited the save_model method of the FileOperation class
22-Mar-22 21:32:41 - ERROR - Unsuccessful end of training
23-Mar-22 09:35:57 - INFO - Start of validation on files for training
23-Mar-22 09:35:58 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
23-Mar-22 09:35:59 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
23-Mar-22 09:35:59 - INFO - Column length validation has been started.
23-Mar-22 09:35:59 - INFO - Column length validation has been completed.
23-Mar-22 09:35:59 - INFO - Missing values validation has been started.
23-Mar-22 09:35:59 - INFO - Raw data validation is completed.
23-Mar-22 09:35:59 - INFO - Creating training database and tables on the basis of given schema
23-Mar-22 09:36:24 - INFO - Opened websites keyspace successfully.
23-Mar-22 09:36:24 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
23-Mar-22 09:36:26 - INFO - Closed websites keyspace successfully.
23-Mar-22 09:36:26 - INFO - Table creation is completed.
23-Mar-22 09:36:26 - INFO - Insertion of data into the database is started.
23-Mar-22 09:36:32 - INFO - Opened websites keyspace successfully.
23-Mar-22 09:37:50 - INFO - The file has been inserted: no need to reinsert.
23-Mar-22 09:37:50 - INFO - Insertion into the database is completed.
23-Mar-22 09:37:50 - INFO - Deleting good data folder
23-Mar-22 09:37:50 - INFO - GoodRaw directory deleted successfully.
23-Mar-22 09:37:50 - INFO - Good_Data folder is deleted
23-Mar-22 09:37:50 - INFO - Moving bad files to archive and deleting Bad_Data folder
23-Mar-22 09:37:50 - INFO - Bad files moved to archive
23-Mar-22 09:37:50 - INFO - Bad Raw Data Folder Deleted successfully.
23-Mar-22 09:37:50 - INFO - Bad files are moved to archive. Bad folder is deleted.
23-Mar-22 09:37:50 - INFO - Validation operation is completed.
23-Mar-22 09:37:50 - INFO - Extracting csv file from the database
23-Mar-22 09:38:00 - INFO - Opened websites keyspace successfully.
23-Mar-22 09:39:15 - INFO - The csv file is exported successfully.
23-Mar-22 09:39:15 - INFO - Start of model training
23-Mar-22 09:39:15 - INFO - Entered the get_data method of the DataGetter class
23-Mar-22 09:39:16 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
23-Mar-22 09:39:16 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 09:39:17 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 09:39:17 - INFO - Entered the is_null_present method of the Preprocessor class
23-Mar-22 09:39:17 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
23-Mar-22 09:39:18 - INFO - Entered the replace_missing_values method of the Preprocessor class
23-Mar-22 09:39:23 - INFO - The -1 values have been replaced with -999.
23-Mar-22 09:39:23 - INFO - Entered the separate_label_feature method of the Preprocessor class
23-Mar-22 09:39:23 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
23-Mar-22 09:39:24 - INFO - Entered the elbow_plot method of the KMeansClustering class
23-Mar-22 09:40:08 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
23-Mar-22 09:40:08 - INFO - Entered the create_clusters method of the KMeansClustering class
23-Mar-22 09:40:09 - INFO - Entered the save_model method of the FileOperation class
23-Mar-22 09:40:09 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
23-Mar-22 09:40:09 - ERROR - Exception occurred in create_clusters method of the KMeansClustering class.             Exception message: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
23-Mar-22 09:40:09 - INFO - Fitting the data to clusters failed. Exited the create_clusters method of             the KMeansClustering class
23-Mar-22 09:40:09 - ERROR - Unsuccessful end of training
23-Mar-22 09:52:13 - INFO - Start of validation on files for training
23-Mar-22 09:52:13 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
23-Mar-22 09:52:14 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
23-Mar-22 09:52:14 - INFO - Column length validation has been started.
23-Mar-22 09:52:14 - INFO - Column length validation has been completed.
23-Mar-22 09:52:14 - INFO - Missing values validation has been started.
23-Mar-22 09:52:14 - INFO - Raw data validation is completed.
23-Mar-22 09:52:14 - INFO - Creating training database and tables on the basis of given schema
23-Mar-22 09:52:29 - INFO - Opened websites keyspace successfully.
23-Mar-22 09:52:29 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
23-Mar-22 09:52:30 - INFO - Closed websites keyspace successfully.
23-Mar-22 09:52:30 - INFO - Table creation is completed.
23-Mar-22 09:52:30 - INFO - Insertion of data into the database is started.
23-Mar-22 09:52:37 - INFO - Opened websites keyspace successfully.
23-Mar-22 09:54:00 - INFO - The file has been inserted: no need to reinsert.
23-Mar-22 09:54:00 - INFO - Insertion into the database is completed.
23-Mar-22 09:54:00 - INFO - Deleting good data folder
23-Mar-22 09:54:00 - INFO - GoodRaw directory deleted successfully.
23-Mar-22 09:54:00 - INFO - Good_Data folder is deleted
23-Mar-22 09:54:00 - INFO - Moving bad files to archive and deleting Bad_Data folder
23-Mar-22 09:54:00 - INFO - Bad files moved to archive
23-Mar-22 09:54:00 - INFO - Bad Raw Data Folder Deleted successfully.
23-Mar-22 09:54:00 - INFO - Bad files are moved to archive. Bad folder is deleted.
23-Mar-22 09:54:00 - INFO - Validation operation is completed.
23-Mar-22 09:54:00 - INFO - Extracting csv file from the database
23-Mar-22 09:54:09 - INFO - Opened websites keyspace successfully.
23-Mar-22 09:55:30 - INFO - The csv file is exported successfully.
23-Mar-22 09:55:30 - INFO - Start of model training
23-Mar-22 09:55:30 - INFO - Entered the get_data method of the DataGetter class
23-Mar-22 09:55:32 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
23-Mar-22 09:55:32 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 09:55:33 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 09:55:33 - INFO - Entered the is_null_present method of the Preprocessor class
23-Mar-22 09:55:33 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
23-Mar-22 09:55:34 - INFO - Entered the replace_missing_values method of the Preprocessor class
23-Mar-22 09:55:39 - INFO - The -1 values have been replaced with -999.
23-Mar-22 09:55:39 - INFO - Entered the separate_label_feature method of the Preprocessor class
23-Mar-22 09:55:39 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
23-Mar-22 09:55:42 - INFO - Entered the elbow_plot method of the KMeansClustering class
23-Mar-22 09:57:02 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
23-Mar-22 09:57:02 - INFO - Entered the create_clusters method of the KMeansClustering class
23-Mar-22 09:57:03 - INFO - Entered the save_model method of the FileOperation class
23-Mar-22 09:57:04 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
23-Mar-22 09:57:04 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
23-Mar-22 09:57:04 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 09:57:05 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 09:57:05 - INFO - Entered the standardize_data method of the Preprocessor class
23-Mar-22 09:57:05 - INFO - The data has been standardized
23-Mar-22 09:57:05 - INFO - Entered the get_best_model method of the ModelFinder class
23-Mar-22 09:57:05 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 10:03:05 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 9, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 10:03:05 - INFO - ROC-AUC for XGBoost: 0.7460335750055626
23-Mar-22 10:03:05 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 10:16:19 - INFO - SVM best params: {'C': 0.1, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 10:16:21 - INFO - ROC-AUC for SVM: 0.5
23-Mar-22 10:16:21 - INFO - Entered the save_model method of the FileOperation class
23-Mar-22 10:16:22 - INFO - Model File XGBoost1 saved. Exited the save_model method of the FileOperation class
23-Mar-22 10:16:30 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 10:16:30 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 10:16:30 - INFO - Entered the standardize_data method of the Preprocessor class
23-Mar-22 10:16:31 - INFO - The data has been standardized
23-Mar-22 10:16:31 - INFO - Entered the get_best_model method of the ModelFinder class
23-Mar-22 10:16:31 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 10:26:40 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 9, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 10:26:40 - INFO - ROC-AUC for XGBoost: 0.9368959659710809
23-Mar-22 10:26:40 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 13:37:35 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 13:38:01 - INFO - ROC-AUC for SVM: 0.7816206088561735
23-Mar-22 13:38:02 - INFO - Entered the save_model method of the FileOperation class
23-Mar-22 13:38:03 - INFO - Model File XGBoost0 saved. Exited the save_model method of the FileOperation class
23-Mar-22 13:38:04 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 13:38:05 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 13:38:05 - INFO - Entered the standardize_data method of the Preprocessor class
23-Mar-22 13:38:06 - INFO - The data has been standardized
23-Mar-22 13:38:06 - INFO - Entered the get_best_model method of the ModelFinder class
23-Mar-22 13:38:06 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 13:39:56 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 9, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 13:39:56 - INFO - ROC-AUC for XGBoost: 0.9294258501074326
23-Mar-22 13:39:56 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 13:42:31 - INFO - SVM best params: {'C': 0.1, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 13:42:31 - INFO - ROC-AUC for SVM: 0.5
23-Mar-22 13:42:32 - INFO - Entered the save_model method of the FileOperation class
23-Mar-22 13:42:32 - INFO - Model File XGBoost2 saved. Exited the save_model method of the FileOperation class
23-Mar-22 13:42:32 - ERROR - Unsuccessful end of training
23-Mar-22 14:59:29 - INFO - Start of validation on files for training
23-Mar-22 14:59:29 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
23-Mar-22 14:59:30 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
23-Mar-22 14:59:30 - INFO - Column length validation has been started.
23-Mar-22 14:59:30 - INFO - Column length validation has been completed.
23-Mar-22 14:59:30 - INFO - Missing values validation has been started.
23-Mar-22 14:59:30 - INFO - Raw data validation is completed.
23-Mar-22 14:59:30 - INFO - Creating training database and tables on the basis of given schema
23-Mar-22 14:59:45 - INFO - Opened websites keyspace successfully.
23-Mar-22 14:59:45 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
23-Mar-22 14:59:45 - INFO - Closed websites keyspace successfully.
23-Mar-22 14:59:45 - INFO - Table creation is completed.
23-Mar-22 14:59:45 - INFO - Insertion of data into the database is started.
23-Mar-22 14:59:53 - INFO - Opened websites keyspace successfully.
23-Mar-22 15:04:44 - INFO - The file has been inserted: no need to reinsert.
23-Mar-22 15:04:44 - INFO - Insertion into the database is completed.
23-Mar-22 15:04:44 - INFO - Deleting good data folder
23-Mar-22 15:04:44 - INFO - GoodRaw directory deleted successfully.
23-Mar-22 15:04:44 - INFO - Good_Data folder is deleted
23-Mar-22 15:04:44 - INFO - Moving bad files to archive and deleting Bad_Data folder
23-Mar-22 15:04:44 - INFO - Bad files moved to archive
23-Mar-22 15:04:44 - INFO - Bad Raw Data Folder Deleted successfully.
23-Mar-22 15:04:44 - INFO - Bad files are moved to archive. Bad folder is deleted.
23-Mar-22 15:04:44 - INFO - Validation operation is completed.
23-Mar-22 15:04:44 - INFO - Extracting csv file from the database
23-Mar-22 15:04:54 - INFO - Opened websites keyspace successfully.
23-Mar-22 15:09:06 - INFO - The csv file is exported successfully.
23-Mar-22 15:09:06 - INFO - Start of model training
23-Mar-22 15:09:06 - INFO - Entered the get_data method of the DataGetter class
23-Mar-22 15:09:08 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
23-Mar-22 15:09:08 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 15:09:11 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 15:09:11 - INFO - Entered the is_null_present method of the Preprocessor class
23-Mar-22 15:09:11 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
23-Mar-22 15:09:12 - INFO - Entered the replace_missing_values method of the Preprocessor class
23-Mar-22 15:09:17 - INFO - The -1 values have been replaced with -999.
23-Mar-22 15:09:17 - INFO - Entered the separate_label_feature method of the Preprocessor class
23-Mar-22 15:09:17 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
23-Mar-22 15:09:17 - INFO - Entered the elbow_plot method of the KMeansClustering class
23-Mar-22 15:09:57 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
23-Mar-22 15:09:57 - INFO - Entered the create_clusters method of the KMeansClustering class
23-Mar-22 15:09:58 - INFO - Entered the save_model method of the FileOperation class
23-Mar-22 15:09:58 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
23-Mar-22 15:09:59 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
23-Mar-22 15:09:59 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 15:09:59 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 15:09:59 - INFO - Entered the standardize_data method of the Preprocessor class
23-Mar-22 15:10:00 - INFO - The data has been standardized
23-Mar-22 15:10:00 - INFO - Entered the get_best_model method of the ModelFinder class
23-Mar-22 15:10:00 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 15:15:56 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 9, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 15:15:56 - INFO - ROC-AUC for XGBoost: 0.7460335750055626
23-Mar-22 15:15:56 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 15:29:48 - INFO - SVM best params: {'C': 0.1, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 15:29:54 - INFO - ROC-AUC for SVM: 0.5
23-Mar-22 15:29:58 - INFO - Entered the save_model method of the FileOperation class
23-Mar-22 15:30:09 - INFO - Model File XGBoost1 saved. Exited the save_model method of the FileOperation class
23-Mar-22 15:30:22 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 15:30:30 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 15:30:30 - INFO - Entered the standardize_data method of the Preprocessor class
23-Mar-22 15:30:33 - INFO - The data has been standardized
23-Mar-22 15:30:33 - INFO - Entered the get_best_model method of the ModelFinder class
23-Mar-22 15:30:33 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 15:40:05 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 9, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 15:40:05 - INFO - ROC-AUC for XGBoost: 0.9368959659710809
23-Mar-22 15:40:05 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 18:59:56 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 19:00:28 - INFO - ROC-AUC for SVM: 0.7816206088561735
23-Mar-22 19:00:30 - INFO - Entered the save_model method of the FileOperation class
23-Mar-22 19:00:32 - INFO - Model File XGBoost0 saved. Exited the save_model method of the FileOperation class
23-Mar-22 19:00:36 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 19:00:40 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
23-Mar-22 19:00:40 - INFO - Entered the standardize_data method of the Preprocessor class
23-Mar-22 19:00:40 - INFO - The data has been standardized
23-Mar-22 19:00:40 - INFO - Entered the get_best_model method of the ModelFinder class
23-Mar-22 19:00:40 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 19:02:44 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 9, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
23-Mar-22 19:02:44 - INFO - ROC-AUC for XGBoost: 0.9294258501074326
23-Mar-22 19:02:44 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 19:05:21 - INFO - SVM best params: {'C': 0.1, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
23-Mar-22 19:05:22 - INFO - ROC-AUC for SVM: 0.5
23-Mar-22 19:05:22 - INFO - Entered the save_model method of the FileOperation class
23-Mar-22 19:05:23 - INFO - Model File XGBoost2 saved. Exited the save_model method of the FileOperation class
23-Mar-22 19:05:23 - ERROR - Unsuccessful end of training
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_model.py", line 141, in training_model
    pickle.dump(drop_pipeline_dict, file)
AttributeError: Can't pickle local object 'Preprocessor.standardize_data.<locals>.log_transform'
24-Mar-22 09:21:47 - INFO - Start of validation on files for training
24-Mar-22 09:21:47 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
24-Mar-22 09:21:48 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
24-Mar-22 09:21:48 - INFO - Column length validation has been started.
24-Mar-22 09:21:48 - INFO - Column length validation has been completed.
24-Mar-22 09:21:48 - INFO - Missing values validation has been started.
24-Mar-22 09:21:48 - INFO - Raw data validation is completed.
24-Mar-22 09:21:48 - INFO - Creating training database and tables on the basis of given schema
24-Mar-22 09:22:16 - INFO - Opened websites keyspace successfully.
24-Mar-22 09:22:16 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
24-Mar-22 09:22:18 - INFO - Closed websites keyspace successfully.
24-Mar-22 09:22:18 - INFO - Table creation is completed.
24-Mar-22 09:22:18 - INFO - Insertion of data into the database is started.
24-Mar-22 09:22:27 - INFO - Opened websites keyspace successfully.
24-Mar-22 09:25:10 - INFO - The file has been inserted: no need to reinsert.
24-Mar-22 09:25:10 - INFO - Insertion into the database is completed.
24-Mar-22 09:25:10 - INFO - Deleting good data folder
24-Mar-22 09:25:10 - INFO - GoodRaw directory deleted successfully.
24-Mar-22 09:25:10 - INFO - Good_Data folder is deleted
24-Mar-22 09:25:10 - INFO - Moving bad files to archive and deleting Bad_Data folder
24-Mar-22 09:25:13 - INFO - Bad files moved to archive
24-Mar-22 09:25:13 - INFO - Bad Raw Data Folder Deleted successfully.
24-Mar-22 09:25:13 - INFO - Bad files are moved to archive. Bad folder is deleted.
24-Mar-22 09:25:13 - INFO - Validation operation is completed.
24-Mar-22 09:25:13 - INFO - Extracting csv file from the database
24-Mar-22 09:25:23 - INFO - Opened websites keyspace successfully.
24-Mar-22 09:28:08 - INFO - The csv file is exported successfully.
24-Mar-22 09:28:08 - INFO - Start of model training
24-Mar-22 09:28:08 - INFO - Entered the get_data method of the DataGetter class
24-Mar-22 09:28:10 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
24-Mar-22 09:28:10 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 09:28:11 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 09:28:11 - INFO - Entered the is_null_present method of the Preprocessor class
24-Mar-22 09:28:11 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
24-Mar-22 09:28:12 - INFO - Entered the replace_missing_values method of the Preprocessor class
24-Mar-22 09:28:17 - INFO - The -1 values have been replaced with -999.
24-Mar-22 09:28:17 - INFO - Entered the separate_label_feature method of the Preprocessor class
24-Mar-22 09:28:17 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
24-Mar-22 09:28:18 - INFO - Entered the elbow_plot method of the KMeansClustering class
24-Mar-22 09:29:40 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
24-Mar-22 09:29:40 - INFO - Entered the create_clusters method of the KMeansClustering class
24-Mar-22 09:29:42 - INFO - Entered the save_model method of the FileOperation class
24-Mar-22 09:29:42 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
24-Mar-22 09:29:43 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
24-Mar-22 09:29:43 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 09:29:44 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 09:29:44 - INFO - Entered the standardize_data method of the Preprocessor class
24-Mar-22 09:29:46 - INFO - The data has been standardized
24-Mar-22 09:29:46 - ERROR - Unsuccessful end of training
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_model.py", line 131, in training_model
    X_test[cont_col_list] = standardization_scaler.transform(X_test[cont_col_list])
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\preprocessing\_data.py", line 980, in transform
    force_all_finite="allow-nan",
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\base.py", line 585, in _validate_data
    self._check_n_features(X, reset=reset)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\base.py", line 401, in _check_n_features
    f"X has {n_features} features, but {self.__class__.__name__} "
ValueError: X has 19 features, but StandardScaler is expecting 18 features as input.
24-Mar-22 09:41:00 - INFO - Start of validation on files for training
24-Mar-22 09:41:01 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
24-Mar-22 09:41:02 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
24-Mar-22 09:41:02 - INFO - Column length validation has been started.
24-Mar-22 09:41:02 - INFO - Column length validation has been completed.
24-Mar-22 09:41:02 - INFO - Missing values validation has been started.
24-Mar-22 09:41:02 - INFO - Raw data validation is completed.
24-Mar-22 09:41:02 - INFO - Creating training database and tables on the basis of given schema
24-Mar-22 09:41:17 - INFO - Opened websites keyspace successfully.
24-Mar-22 09:41:17 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
24-Mar-22 09:41:19 - INFO - Closed websites keyspace successfully.
24-Mar-22 09:41:19 - INFO - Table creation is completed.
24-Mar-22 09:41:19 - INFO - Insertion of data into the database is started.
24-Mar-22 09:41:27 - INFO - Opened websites keyspace successfully.
24-Mar-22 09:42:52 - INFO - The file has been inserted: no need to reinsert.
24-Mar-22 09:42:52 - INFO - Insertion into the database is completed.
24-Mar-22 09:42:52 - INFO - Deleting good data folder
24-Mar-22 09:42:52 - INFO - GoodRaw directory deleted successfully.
24-Mar-22 09:42:52 - INFO - Good_Data folder is deleted
24-Mar-22 09:42:52 - INFO - Moving bad files to archive and deleting Bad_Data folder
24-Mar-22 09:42:52 - INFO - Bad files moved to archive
24-Mar-22 09:42:52 - INFO - Bad Raw Data Folder Deleted successfully.
24-Mar-22 09:42:52 - INFO - Bad files are moved to archive. Bad folder is deleted.
24-Mar-22 09:42:52 - INFO - Validation operation is completed.
24-Mar-22 09:42:52 - INFO - Extracting csv file from the database
24-Mar-22 09:43:00 - INFO - Opened websites keyspace successfully.
24-Mar-22 09:44:20 - INFO - The csv file is exported successfully.
24-Mar-22 09:44:20 - INFO - Start of model training
24-Mar-22 09:44:20 - INFO - Entered the get_data method of the DataGetter class
24-Mar-22 09:44:22 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
24-Mar-22 09:44:22 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 09:44:24 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 09:44:24 - INFO - Entered the is_null_present method of the Preprocessor class
24-Mar-22 09:44:24 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
24-Mar-22 09:44:25 - INFO - Entered the replace_missing_values method of the Preprocessor class
24-Mar-22 09:44:31 - INFO - The -1 values have been replaced with -999.
24-Mar-22 09:44:31 - INFO - Entered the separate_label_feature method of the Preprocessor class
24-Mar-22 09:44:31 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
24-Mar-22 09:44:32 - INFO - Entered the elbow_plot method of the KMeansClustering class
24-Mar-22 09:45:13 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
24-Mar-22 09:45:13 - INFO - Entered the create_clusters method of the KMeansClustering class
24-Mar-22 09:45:14 - INFO - Entered the save_model method of the FileOperation class
24-Mar-22 09:45:14 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
24-Mar-22 09:45:14 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
24-Mar-22 09:45:15 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 09:45:15 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 09:45:15 - INFO - Entered the standardize_data method of the Preprocessor class
24-Mar-22 09:45:17 - INFO - The data has been standardized
24-Mar-22 09:45:18 - INFO - Entered the get_best_model method of the ModelFinder class
24-Mar-22 09:45:18 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 09:49:27 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 8, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 09:49:27 - INFO - ROC-AUC for XGBoost: 0.7527903317623194
24-Mar-22 09:49:27 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 10:02:09 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 10:02:12 - INFO - ROC-AUC for SVM: 0.5
24-Mar-22 10:02:13 - INFO - Entered the save_model method of the FileOperation class
24-Mar-22 10:02:13 - INFO - Model File XGBoost1 saved. Exited the save_model method of the FileOperation class
24-Mar-22 10:02:17 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 10:02:19 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 10:02:19 - INFO - Entered the standardize_data method of the Preprocessor class
24-Mar-22 10:02:26 - INFO - The data has been standardized
24-Mar-22 10:02:28 - INFO - Entered the get_best_model method of the ModelFinder class
24-Mar-22 10:02:28 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 10:08:56 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 8, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 10:08:56 - INFO - ROC-AUC for XGBoost: 0.9349341165386131
24-Mar-22 10:08:56 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 12:01:21 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 12:01:39 - INFO - ROC-AUC for SVM: 0.8478046774418773
24-Mar-22 12:01:44 - INFO - Entered the save_model method of the FileOperation class
24-Mar-22 12:01:49 - INFO - Model File XGBoost0 saved. Exited the save_model method of the FileOperation class
24-Mar-22 12:02:01 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 12:02:03 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 12:02:03 - INFO - Entered the standardize_data method of the Preprocessor class
24-Mar-22 12:02:06 - INFO - The data has been standardized
24-Mar-22 12:02:06 - INFO - Entered the get_best_model method of the ModelFinder class
24-Mar-22 12:02:06 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 12:03:27 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 9, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 12:03:27 - INFO - ROC-AUC for XGBoost: 0.932651501384331
24-Mar-22 12:03:27 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 12:05:21 - INFO - SVM best params: {'C': 0.1, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 12:05:22 - INFO - ROC-AUC for SVM: 0.5
24-Mar-22 12:05:22 - INFO - Entered the save_model method of the FileOperation class
24-Mar-22 12:05:22 - INFO - Model File XGBoost2 saved. Exited the save_model method of the FileOperation class
24-Mar-22 12:05:22 - INFO - Successful end of training
24-Mar-22 14:58:33 - INFO - Start of validation on files for training
24-Mar-22 14:58:33 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
24-Mar-22 14:58:34 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
24-Mar-22 14:58:34 - INFO - Column length validation has been started.
24-Mar-22 14:58:34 - INFO - Column length validation has been completed.
24-Mar-22 14:58:34 - INFO - Missing values validation has been started.
24-Mar-22 14:58:34 - INFO - Raw data validation is completed.
24-Mar-22 14:58:34 - INFO - Creating training database and tables on the basis of given schema
24-Mar-22 14:58:41 - ERROR - Error while creating tables: Unable to connect to the metadata service at https://5ab86f0e-9ab1-4cb8-9ae5-006ee8cc4a37-europe-west1.db.astra.datastax.com:29080/metadata. Check the cluster status in the cloud console.  
24-Mar-22 15:03:31 - INFO - Start of validation on files for training
24-Mar-22 15:03:31 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
24-Mar-22 15:03:31 - INFO - Bad_Raw directory deleted before starting validation.
24-Mar-22 15:03:31 - INFO - GoodRaw directory deleted successfully.
24-Mar-22 15:03:32 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
24-Mar-22 15:03:32 - INFO - Column length validation has been started.
24-Mar-22 15:03:32 - INFO - Column length validation has been completed.
24-Mar-22 15:03:32 - INFO - Missing values validation has been started.
24-Mar-22 15:03:32 - INFO - Raw data validation is completed.
24-Mar-22 15:03:32 - INFO - Creating training database and tables on the basis of given schema
24-Mar-22 15:04:02 - INFO - Opened websites keyspace successfully.
24-Mar-22 15:04:02 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
24-Mar-22 15:04:02 - INFO - Closed websites keyspace successfully.
24-Mar-22 15:04:02 - INFO - Table creation is completed.
24-Mar-22 15:04:02 - INFO - Insertion of data into the database is started.
24-Mar-22 15:04:12 - INFO - Opened websites keyspace successfully.
24-Mar-22 15:04:32 - INFO - Insertion into the database is completed.
24-Mar-22 15:04:32 - INFO - Deleting good data folder
24-Mar-22 15:04:32 - INFO - GoodRaw directory deleted successfully.
24-Mar-22 15:04:32 - INFO - Good_Data folder is deleted
24-Mar-22 15:04:32 - INFO - Moving bad files to archive and deleting Bad_Data folder
24-Mar-22 15:04:33 - INFO - Bad files moved to archive
24-Mar-22 15:04:33 - INFO - Bad Raw Data Folder Deleted successfully.
24-Mar-22 15:04:33 - INFO - Bad files are moved to archive. Bad folder is deleted.
24-Mar-22 15:04:33 - INFO - Validation operation is completed.
24-Mar-22 15:04:33 - INFO - Extracting csv file from the database
24-Mar-22 15:04:42 - INFO - Opened websites keyspace successfully.
24-Mar-22 15:07:29 - INFO - The csv file is exported successfully.
24-Mar-22 15:07:29 - INFO - Start of model training
24-Mar-22 15:07:29 - INFO - Entered the get_data method of the DataGetter class
24-Mar-22 15:07:31 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
24-Mar-22 15:07:31 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 15:07:35 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 15:07:35 - INFO - Entered the is_null_present method of the Preprocessor class
24-Mar-22 15:07:35 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
24-Mar-22 15:07:36 - INFO - Entered the replace_missing_values method of the Preprocessor class
24-Mar-22 15:07:40 - INFO - The -1 values have been replaced with -999.
24-Mar-22 15:07:40 - INFO - Entered the separate_label_feature method of the Preprocessor class
24-Mar-22 15:07:40 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
24-Mar-22 15:07:41 - INFO - Entered the elbow_plot method of the KMeansClustering class
24-Mar-22 15:08:36 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
24-Mar-22 15:08:36 - INFO - Entered the create_clusters method of the KMeansClustering class
24-Mar-22 15:08:38 - INFO - Entered the save_model method of the FileOperation class
24-Mar-22 15:08:38 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
24-Mar-22 15:08:38 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
24-Mar-22 15:08:39 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 15:08:40 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 15:08:40 - INFO - Entered the standardize_data method of the Preprocessor class
24-Mar-22 15:08:41 - INFO - The data has been standardized
24-Mar-22 15:08:42 - INFO - Entered the get_best_model method of the ModelFinder class
24-Mar-22 15:08:42 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 15:13:06 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 8, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 15:13:06 - INFO - ROC-AUC for XGBoost: 0.7527903317623194
24-Mar-22 15:13:06 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 15:25:28 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 15:25:31 - INFO - ROC-AUC for SVM: 0.5
24-Mar-22 15:25:31 - INFO - Entered the save_model method of the FileOperation class
24-Mar-22 15:25:31 - INFO - Model File XGBoost1 saved. Exited the save_model method of the FileOperation class
24-Mar-22 15:25:31 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 15:25:32 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 15:25:32 - INFO - Entered the standardize_data method of the Preprocessor class
24-Mar-22 15:25:36 - INFO - The data has been standardized
24-Mar-22 15:25:37 - INFO - Entered the get_best_model method of the ModelFinder class
24-Mar-22 15:25:37 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 15:31:36 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 8, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 15:31:36 - INFO - ROC-AUC for XGBoost: 0.9349341165386131
24-Mar-22 15:31:36 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 17:15:49 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 17:16:04 - INFO - ROC-AUC for SVM: 0.8478046774418773
24-Mar-22 17:16:04 - INFO - Entered the save_model method of the FileOperation class
24-Mar-22 17:16:05 - INFO - Model File XGBoost0 saved. Exited the save_model method of the FileOperation class
24-Mar-22 17:16:05 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 17:16:05 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
24-Mar-22 17:16:05 - INFO - Entered the standardize_data method of the Preprocessor class
24-Mar-22 17:16:06 - INFO - The data has been standardized
24-Mar-22 17:16:07 - INFO - Entered the get_best_model method of the ModelFinder class
24-Mar-22 17:16:07 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 17:17:26 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 9, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
24-Mar-22 17:17:26 - INFO - ROC-AUC for XGBoost: 0.932651501384331
24-Mar-22 17:17:26 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 17:19:21 - INFO - SVM best params: {'C': 0.1, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
24-Mar-22 17:19:22 - INFO - ROC-AUC for SVM: 0.5
24-Mar-22 17:19:22 - INFO - Entered the save_model method of the FileOperation class
24-Mar-22 17:19:22 - INFO - Model File XGBoost2 saved. Exited the save_model method of the FileOperation class
24-Mar-22 17:19:22 - INFO - Successful end of training
28-Mar-22 09:40:11 - INFO - Start of validation on files for training
28-Mar-22 09:40:12 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
28-Mar-22 09:40:22 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
28-Mar-22 09:40:22 - INFO - Column length validation has been started.
28-Mar-22 09:40:22 - INFO - Column length validation has been completed.
28-Mar-22 09:40:22 - INFO - Missing values validation has been started.
28-Mar-22 09:40:22 - INFO - Raw data validation is completed.
28-Mar-22 09:40:22 - INFO - Creating training database and tables on the basis of given schema
28-Mar-22 09:40:46 - ERROR - Error while creating tables: Unable to connect to the metadata service at https://5ab86f0e-9ab1-4cb8-9ae5-006ee8cc4a37-europe-west1.db.astra.datastax.com:29080/metadata. Check the cluster status in the cloud console.  
28-Mar-22 09:44:52 - INFO - Start of validation on files for training
28-Mar-22 09:44:52 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
28-Mar-22 09:44:53 - INFO - Bad_Raw directory deleted before starting validation.
28-Mar-22 09:44:53 - INFO - GoodRaw directory deleted successfully.
28-Mar-22 09:44:53 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
28-Mar-22 09:44:53 - INFO - Column length validation has been started.
28-Mar-22 09:44:53 - INFO - Column length validation has been completed.
28-Mar-22 09:44:53 - INFO - Missing values validation has been started.
28-Mar-22 09:44:53 - INFO - Raw data validation is completed.
28-Mar-22 09:44:53 - INFO - Creating training database and tables on the basis of given schema
28-Mar-22 09:45:06 - INFO - Opened websites keyspace successfully.
28-Mar-22 09:45:06 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
28-Mar-22 09:45:07 - INFO - Closed websites keyspace successfully.
28-Mar-22 09:45:07 - INFO - Table creation is completed.
28-Mar-22 09:45:07 - INFO - Insertion of data into the database is started.
28-Mar-22 09:45:16 - INFO - Opened websites keyspace successfully.
28-Mar-22 09:48:18 - INFO - The file has been inserted: no need to reinsert.
28-Mar-22 09:48:18 - INFO - Insertion into the database is completed.
28-Mar-22 09:48:18 - INFO - Deleting good data folder
28-Mar-22 09:48:18 - INFO - GoodRaw directory deleted successfully.
28-Mar-22 09:48:18 - INFO - Good_Data folder is deleted
28-Mar-22 09:48:18 - INFO - Moving bad files to archive and deleting Bad_Data folder
28-Mar-22 09:48:18 - INFO - Bad files moved to archive
28-Mar-22 09:48:18 - INFO - Bad Raw Data Folder Deleted successfully.
28-Mar-22 09:48:18 - INFO - Bad files are moved to archive. Bad folder is deleted.
28-Mar-22 09:48:18 - INFO - Validation operation is completed.
28-Mar-22 09:48:18 - INFO - Extracting csv file from the database
28-Mar-22 09:48:28 - INFO - Opened websites keyspace successfully.
28-Mar-22 09:51:20 - INFO - The csv file is exported successfully.
28-Mar-22 09:51:20 - INFO - Start of model training
28-Mar-22 09:51:20 - INFO - Entered the get_data method of the DataGetter class
28-Mar-22 09:51:22 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
28-Mar-22 09:51:22 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 09:51:24 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 09:51:24 - INFO - Entered the is_null_present method of the Preprocessor class
28-Mar-22 09:51:24 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
28-Mar-22 09:51:25 - INFO - Entered the replace_missing_values method of the Preprocessor class
28-Mar-22 09:51:30 - INFO - The -1 values have been replaced with -999.
28-Mar-22 09:51:30 - INFO - Entered the separate_label_feature method of the Preprocessor class
28-Mar-22 09:51:30 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
28-Mar-22 09:51:31 - INFO - Entered the elbow_plot method of the KMeansClustering class
28-Mar-22 09:52:41 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
28-Mar-22 09:52:41 - INFO - Entered the create_clusters method of the KMeansClustering class
28-Mar-22 09:52:43 - INFO - Entered the save_model method of the FileOperation class
28-Mar-22 09:52:43 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
28-Mar-22 09:52:43 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
28-Mar-22 09:52:44 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 09:52:45 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 09:52:45 - INFO - Entered the standardize_data method of the Preprocessor class
28-Mar-22 09:52:47 - INFO - The data has been standardized
28-Mar-22 09:52:47 - INFO - Entered the get_best_model method of the ModelFinder class
28-Mar-22 09:52:47 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 09:57:38 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 8, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 09:57:38 - INFO - ROC-AUC for XGBoost: 0.7527903317623194
28-Mar-22 09:57:38 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 10:10:02 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 10:10:05 - INFO - ROC-AUC for SVM: 0.5
28-Mar-22 10:10:05 - INFO - Entered the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 10:10:43 - INFO - LogisticRegression best params: {'C': 0.01, 'random_state': 0}             Exited the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 10:10:43 - INFO - ROC-AUC for LogisticRegression: 0.5067032979734787
28-Mar-22 10:10:43 - ERROR - Exception occurred in get_best_model method of the ModelFinder class.             Exception message: X has 19 features, but GaussianNB is expecting 10 features as input.
28-Mar-22 10:10:43 - INFO - Model selection Failed. Exited the get_best_model method of the Model_Finder class
28-Mar-22 10:10:43 - ERROR - Unsuccessful end of training
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 390, in get_best_model
    'gnb': self.gnb.predict_proba(train_x[cont_col_list])[:, 1],
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\naive_bayes.py", line 125, in predict_proba
    return np.exp(self.predict_log_proba(X))
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\naive_bayes.py", line 103, in predict_log_proba
    X = self._check_X(X)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\naive_bayes.py", line 251, in _check_X
    return self._validate_data(X, reset=False)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\base.py", line 585, in _validate_data
    self._check_n_features(X, reset=reset)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\base.py", line 401, in _check_n_features
    f"X has {n_features} features, but {self.__class__.__name__} "
ValueError: X has 19 features, but GaussianNB is expecting 10 features as input.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_model.py", line 142, in training_model
    best_model_name, best_model = model_finder.get_best_model(X_train, y_train, X_test,  y_test)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 486, in get_best_model
    raise Exception()
Exception
28-Mar-22 11:30:31 - INFO - Start of validation on files for training
28-Mar-22 11:30:31 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
28-Mar-22 11:30:32 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
28-Mar-22 11:30:32 - INFO - Column length validation has been started.
28-Mar-22 11:30:32 - INFO - Column length validation has been completed.
28-Mar-22 11:30:32 - INFO - Missing values validation has been started.
28-Mar-22 11:30:32 - INFO - Raw data validation is completed.
28-Mar-22 11:30:32 - INFO - Creating training database and tables on the basis of given schema
28-Mar-22 11:30:46 - INFO - Opened websites keyspace successfully.
28-Mar-22 11:30:46 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
28-Mar-22 11:30:48 - INFO - Closed websites keyspace successfully.
28-Mar-22 11:30:48 - INFO - Table creation is completed.
28-Mar-22 11:30:48 - INFO - Insertion of data into the database is started.
28-Mar-22 11:30:57 - INFO - Opened websites keyspace successfully.
28-Mar-22 11:32:25 - INFO - The file has been inserted: no need to reinsert.
28-Mar-22 11:32:25 - INFO - Insertion into the database is completed.
28-Mar-22 11:32:25 - INFO - Deleting good data folder
28-Mar-22 11:32:25 - INFO - GoodRaw directory deleted successfully.
28-Mar-22 11:32:25 - INFO - Good_Data folder is deleted
28-Mar-22 11:32:25 - INFO - Moving bad files to archive and deleting Bad_Data folder
28-Mar-22 11:32:25 - INFO - Bad files moved to archive
28-Mar-22 11:32:25 - INFO - Bad Raw Data Folder Deleted successfully.
28-Mar-22 11:32:25 - INFO - Bad files are moved to archive. Bad folder is deleted.
28-Mar-22 11:32:25 - INFO - Validation operation is completed.
28-Mar-22 11:32:25 - INFO - Extracting csv file from the database
28-Mar-22 11:32:33 - INFO - Opened websites keyspace successfully.
28-Mar-22 11:34:00 - INFO - The csv file is exported successfully.
28-Mar-22 11:34:00 - INFO - Start of model training
28-Mar-22 11:34:00 - INFO - Entered the get_data method of the DataGetter class
28-Mar-22 11:34:02 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
28-Mar-22 11:34:02 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 11:34:03 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 11:34:03 - INFO - Entered the is_null_present method of the Preprocessor class
28-Mar-22 11:34:03 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
28-Mar-22 11:34:04 - INFO - Entered the replace_missing_values method of the Preprocessor class
28-Mar-22 11:34:09 - INFO - The -1 values have been replaced with -999.
28-Mar-22 11:34:09 - INFO - Entered the separate_label_feature method of the Preprocessor class
28-Mar-22 11:34:09 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
28-Mar-22 11:34:10 - INFO - Entered the elbow_plot method of the KMeansClustering class
28-Mar-22 11:34:54 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
28-Mar-22 11:34:54 - INFO - Entered the create_clusters method of the KMeansClustering class
28-Mar-22 11:34:55 - INFO - Entered the save_model method of the FileOperation class
28-Mar-22 11:34:55 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
28-Mar-22 11:34:55 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
28-Mar-22 11:34:56 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 11:34:56 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 11:34:56 - INFO - Entered the standardize_data method of the Preprocessor class
28-Mar-22 11:34:58 - INFO - The data has been standardized
28-Mar-22 11:34:58 - INFO - Entered the get_best_model method of the ModelFinder class
28-Mar-22 11:34:58 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 11:39:16 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 8, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 11:39:16 - INFO - ROC-AUC for XGBoost: 0.7527903317623194
28-Mar-22 11:39:16 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 11:51:52 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 11:51:56 - INFO - ROC-AUC for SVM: 0.5
28-Mar-22 11:51:56 - INFO - Entered the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 11:52:34 - INFO - LogisticRegression best params: {'C': 0.01, 'random_state': 0}             Exited the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 11:52:34 - INFO - ROC-AUC for LogisticRegression: 0.5067032979734787
28-Mar-22 11:52:35 - INFO - ROC-AUC for MixedNaiveBayes: 0.5011760932321181
28-Mar-22 11:52:35 - INFO - Entered the get_best_params_for_rf method of the ModelFinder class
28-Mar-22 11:55:29 - ERROR - Exception occured in get_best_params_for_rf method of the ModelFinder class.             Exception message: Input contains NaN, infinity or a value too large for dtype('float64').
28-Mar-22 11:55:29 - INFO - RandomForest training failed. Exited the get_best_params_for_rf method of             the ModelFinder class
28-Mar-22 11:55:29 - ERROR - Exception occurred in get_best_model method of the ModelFinder class.             Exception message: 
28-Mar-22 11:55:29 - INFO - Model selection Failed. Exited the get_best_model method of the Model_Finder class
28-Mar-22 11:55:29 - ERROR - Unsuccessful end of training
Traceback (most recent call last):
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 179, in maximize
    x_probe = next(self._queue)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 25, in __next__
    raise StopIteration("Queue is empty, no more objects to retrieve.")
StopIteration: Queue is empty, no more objects to retrieve.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 167, in get_best_params_for_rf
    rf_bo_result.maximize(init_points=20, n_iter=4)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 182, in maximize
    x_probe = self.suggest(util)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 128, in suggest
    self._gp.fit(self._space.params, self._space.target)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\gaussian_process\_gpr.py", line 230, in fit
    dtype=dtype,
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\base.py", line 581, in _validate_data
    X, y = check_X_y(X, y, **check_params)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 979, in check_X_y
    y = _check_y(y, multi_output=multi_output, y_numeric=y_numeric)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 990, in _check_y
    y, accept_sparse="csr", force_all_finite=True, ensure_2d=False, dtype=None
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 800, in check_array
    _assert_all_finite(array, allow_nan=force_all_finite == "allow-nan")
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 116, in _assert_all_finite
    type_err, msg_dtype if msg_dtype is not None else X.dtype
ValueError: Input contains NaN, infinity or a value too large for dtype('float64').

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 426, in get_best_model
    self.rf = self.get_best_params_for_rf(train_x, train_y)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 188, in get_best_params_for_rf
    raise Exception()
Exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_model.py", line 142, in training_model
    best_model_name, best_model = model_finder.get_best_model(X_train, y_train, X_test,  y_test)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 486, in get_best_model
    raise Exception()
Exception
28-Mar-22 12:04:25 - INFO - Start of validation on files for training
28-Mar-22 12:04:25 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
28-Mar-22 12:04:26 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
28-Mar-22 12:04:26 - INFO - Column length validation has been started.
28-Mar-22 12:04:26 - INFO - Column length validation has been completed.
28-Mar-22 12:04:26 - INFO - Missing values validation has been started.
28-Mar-22 12:04:26 - INFO - Raw data validation is completed.
28-Mar-22 12:04:26 - INFO - Creating training database and tables on the basis of given schema
28-Mar-22 12:04:37 - INFO - Opened websites keyspace successfully.
28-Mar-22 12:04:37 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
28-Mar-22 12:04:38 - INFO - Closed websites keyspace successfully.
28-Mar-22 12:04:38 - INFO - Table creation is completed.
28-Mar-22 12:04:38 - INFO - Insertion of data into the database is started.
28-Mar-22 12:04:45 - INFO - Opened websites keyspace successfully.
28-Mar-22 12:06:24 - INFO - The file has been inserted: no need to reinsert.
28-Mar-22 12:06:24 - INFO - Insertion into the database is completed.
28-Mar-22 12:06:24 - INFO - Deleting good data folder
28-Mar-22 12:06:24 - INFO - GoodRaw directory deleted successfully.
28-Mar-22 12:06:24 - INFO - Good_Data folder is deleted
28-Mar-22 12:06:24 - INFO - Moving bad files to archive and deleting Bad_Data folder
28-Mar-22 12:06:24 - INFO - Bad files moved to archive
28-Mar-22 12:06:24 - INFO - Bad Raw Data Folder Deleted successfully.
28-Mar-22 12:06:24 - INFO - Bad files are moved to archive. Bad folder is deleted.
28-Mar-22 12:06:24 - INFO - Validation operation is completed.
28-Mar-22 12:06:24 - INFO - Extracting csv file from the database
28-Mar-22 12:06:33 - INFO - Opened websites keyspace successfully.
28-Mar-22 12:08:42 - INFO - The csv file is exported successfully.
28-Mar-22 12:08:42 - INFO - Start of model training
28-Mar-22 12:08:42 - INFO - Entered the get_data method of the DataGetter class
28-Mar-22 12:08:43 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
28-Mar-22 12:08:43 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 12:08:44 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 12:08:44 - INFO - Entered the is_null_present method of the Preprocessor class
28-Mar-22 12:08:44 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
28-Mar-22 12:08:44 - INFO - Entered the replace_missing_values method of the Preprocessor class
28-Mar-22 12:08:49 - INFO - The -1 values have been replaced with -999.
28-Mar-22 12:08:49 - INFO - Entered the separate_label_feature method of the Preprocessor class
28-Mar-22 12:08:49 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
28-Mar-22 12:08:50 - INFO - Entered the elbow_plot method of the KMeansClustering class
28-Mar-22 12:09:33 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
28-Mar-22 12:09:33 - INFO - Entered the create_clusters method of the KMeansClustering class
28-Mar-22 12:09:35 - INFO - Entered the save_model method of the FileOperation class
28-Mar-22 12:09:35 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
28-Mar-22 12:09:35 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
28-Mar-22 12:09:35 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 12:09:36 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 12:09:36 - INFO - Entered the standardize_data method of the Preprocessor class
28-Mar-22 12:09:38 - INFO - The data has been standardized
28-Mar-22 12:09:39 - INFO - Entered the get_best_model method of the ModelFinder class
28-Mar-22 12:09:39 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 12:13:51 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 8, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 12:13:52 - INFO - ROC-AUC for XGBoost: 0.7527903317623194
28-Mar-22 12:13:52 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 12:26:09 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 12:26:12 - INFO - ROC-AUC for SVM: 0.5
28-Mar-22 12:26:12 - INFO - Entered the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 12:26:49 - INFO - LogisticRegression best params: {'C': 0.01, 'random_state': 0}             Exited the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 12:26:49 - INFO - ROC-AUC for LogisticRegression: 0.5067032979734787
28-Mar-22 12:26:49 - INFO - ROC-AUC for MixedNaiveBayes: 0.5011760932321181
28-Mar-22 12:26:49 - INFO - Entered the get_best_params_for_lgb method of the ModelFinder class
28-Mar-22 12:28:27 - ERROR - Exception occured in get_best_params_for_lgb method of the ModelFinder class.             Exception message: Input contains NaN, infinity or a value too large for dtype('float64').
28-Mar-22 12:28:27 - INFO - LightGBM training failed. Exited the get_best_params_for_lgb method of             the ModelFinder class
28-Mar-22 12:28:27 - ERROR - Exception occurred in get_best_model method of the ModelFinder class.             Exception message: 
28-Mar-22 12:28:27 - INFO - Model selection Failed. Exited the get_best_model method of the Model_Finder class
28-Mar-22 12:28:27 - ERROR - Unsuccessful end of training
Traceback (most recent call last):
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 179, in maximize
    x_probe = next(self._queue)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 25, in __next__
    raise StopIteration("Queue is empty, no more objects to retrieve.")
StopIteration: Queue is empty, no more objects to retrieve.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 246, in get_best_params_for_lgb
    lgb_bo_result.maximize(init_points=20, n_iter=4)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 182, in maximize
    x_probe = self.suggest(util)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 128, in suggest
    self._gp.fit(self._space.params, self._space.target)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\gaussian_process\_gpr.py", line 230, in fit
    dtype=dtype,
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\base.py", line 581, in _validate_data
    X, y = check_X_y(X, y, **check_params)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 979, in check_X_y
    y = _check_y(y, multi_output=multi_output, y_numeric=y_numeric)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 990, in _check_y
    y, accept_sparse="csr", force_all_finite=True, ensure_2d=False, dtype=None
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 800, in check_array
    _assert_all_finite(array, allow_nan=force_all_finite == "allow-nan")
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 116, in _assert_all_finite
    type_err, msg_dtype if msg_dtype is not None else X.dtype
ValueError: Input contains NaN, infinity or a value too large for dtype('float64').

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 438, in get_best_model
    self.lgb = self.get_best_params_for_lgb(train_x, train_y)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 284, in get_best_params_for_lgb
    raise Exception()
Exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_model.py", line 142, in training_model
    best_model_name, best_model = model_finder.get_best_model(X_train, y_train, X_test,  y_test)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 486, in get_best_model
    raise Exception()
Exception
28-Mar-22 15:31:01 - INFO - Start of validation on files for training
28-Mar-22 15:31:01 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
28-Mar-22 15:31:02 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
28-Mar-22 15:31:02 - INFO - Column length validation has been started.
28-Mar-22 15:31:02 - INFO - Column length validation has been completed.
28-Mar-22 15:31:02 - INFO - Missing values validation has been started.
28-Mar-22 15:31:02 - INFO - Raw data validation is completed.
28-Mar-22 15:31:02 - INFO - Creating training database and tables on the basis of given schema
28-Mar-22 15:31:23 - INFO - Opened websites keyspace successfully.
28-Mar-22 15:31:24 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
28-Mar-22 15:31:26 - INFO - Closed websites keyspace successfully.
28-Mar-22 15:31:26 - INFO - Table creation is completed.
28-Mar-22 15:31:26 - INFO - Insertion of data into the database is started.
28-Mar-22 15:31:38 - INFO - Opened websites keyspace successfully.
28-Mar-22 15:33:41 - INFO - The file has been inserted: no need to reinsert.
28-Mar-22 15:33:41 - INFO - Insertion into the database is completed.
28-Mar-22 15:33:41 - INFO - Deleting good data folder
28-Mar-22 15:33:41 - INFO - GoodRaw directory deleted successfully.
28-Mar-22 15:33:41 - INFO - Good_Data folder is deleted
28-Mar-22 15:33:41 - INFO - Moving bad files to archive and deleting Bad_Data folder
28-Mar-22 15:33:42 - INFO - Bad files moved to archive
28-Mar-22 15:33:42 - INFO - Bad Raw Data Folder Deleted successfully.
28-Mar-22 15:33:42 - INFO - Bad files are moved to archive. Bad folder is deleted.
28-Mar-22 15:33:42 - INFO - Validation operation is completed.
28-Mar-22 15:33:42 - INFO - Extracting csv file from the database
28-Mar-22 15:34:06 - INFO - Opened websites keyspace successfully.
28-Mar-22 15:35:55 - INFO - The csv file is exported successfully.
28-Mar-22 15:35:55 - INFO - Start of model training
28-Mar-22 15:35:55 - INFO - Entered the get_data method of the DataGetter class
28-Mar-22 15:35:57 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
28-Mar-22 15:35:57 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 15:35:59 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 15:35:59 - INFO - Entered the is_null_present method of the Preprocessor class
28-Mar-22 15:35:59 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
28-Mar-22 15:36:00 - INFO - Entered the replace_missing_values method of the Preprocessor class
28-Mar-22 15:36:04 - INFO - The -1 values have been replaced with -999.
28-Mar-22 15:36:04 - INFO - Entered the separate_label_feature method of the Preprocessor class
28-Mar-22 15:36:04 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
28-Mar-22 15:36:07 - INFO - Entered the elbow_plot method of the KMeansClustering class
28-Mar-22 15:37:51 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
28-Mar-22 15:37:51 - INFO - Entered the create_clusters method of the KMeansClustering class
28-Mar-22 15:37:53 - INFO - Entered the save_model method of the FileOperation class
28-Mar-22 15:37:53 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
28-Mar-22 15:37:53 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
28-Mar-22 15:37:55 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 15:37:55 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 15:37:55 - INFO - Entered the standardize_data method of the Preprocessor class
28-Mar-22 15:37:58 - INFO - The data has been standardized
28-Mar-22 15:37:58 - INFO - Entered the get_best_model method of the ModelFinder class
28-Mar-22 15:37:58 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 15:42:25 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 8, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 15:42:25 - INFO - ROC-AUC for XGBoost: 0.7527903317623194
28-Mar-22 15:42:25 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 15:54:45 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 15:54:48 - INFO - ROC-AUC for SVM: 0.5
28-Mar-22 15:54:48 - INFO - Entered the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 15:55:26 - INFO - LogisticRegression best params: {'C': 0.01, 'random_state': 0}             Exited the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 15:55:26 - INFO - ROC-AUC for LogisticRegression: 0.5067032979734787
28-Mar-22 15:55:27 - INFO - ROC-AUC for MixedNaiveBayes: 0.5011760932321181
28-Mar-22 15:55:27 - INFO - Entered the get_best_params_for_rf method of the ModelFinder class
28-Mar-22 15:58:24 - ERROR - Exception occured in get_best_params_for_rf method of the ModelFinder class.             Exception message: Input contains NaN, infinity or a value too large for dtype('float64').
28-Mar-22 15:58:24 - INFO - RandomForest training failed. Exited the get_best_params_for_rf method of             the ModelFinder class
28-Mar-22 15:58:24 - ERROR - Exception occurred in get_best_model method of the ModelFinder class.             Exception message: 
28-Mar-22 15:58:24 - INFO - Model selection Failed. Exited the get_best_model method of the Model_Finder class
28-Mar-22 15:58:24 - ERROR - Unsuccessful end of training
Traceback (most recent call last):
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 179, in maximize
    x_probe = next(self._queue)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 25, in __next__
    raise StopIteration("Queue is empty, no more objects to retrieve.")
StopIteration: Queue is empty, no more objects to retrieve.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 167, in get_best_params_for_rf
    rf_bo_result.maximize(init_points=45, n_iter=20)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 182, in maximize
    x_probe = self.suggest(util)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 128, in suggest
    self._gp.fit(self._space.params, self._space.target)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\gaussian_process\_gpr.py", line 230, in fit
    dtype=dtype,
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\base.py", line 581, in _validate_data
    X, y = check_X_y(X, y, **check_params)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 979, in check_X_y
    y = _check_y(y, multi_output=multi_output, y_numeric=y_numeric)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 990, in _check_y
    y, accept_sparse="csr", force_all_finite=True, ensure_2d=False, dtype=None
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 800, in check_array
    _assert_all_finite(array, allow_nan=force_all_finite == "allow-nan")
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 116, in _assert_all_finite
    type_err, msg_dtype if msg_dtype is not None else X.dtype
ValueError: Input contains NaN, infinity or a value too large for dtype('float64').

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 426, in get_best_model
    self.rf = self.get_best_params_for_rf(train_x, train_y)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 188, in get_best_params_for_rf
    raise Exception()
Exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_model.py", line 142, in training_model
    best_model_name, best_model = model_finder.get_best_model(X_train, y_train, X_test,  y_test)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 486, in get_best_model
    raise Exception()
Exception
28-Mar-22 16:33:02 - INFO - Start of validation on files for training
28-Mar-22 16:33:02 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
28-Mar-22 16:33:03 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
28-Mar-22 16:33:03 - INFO - Column length validation has been started.
28-Mar-22 16:33:03 - INFO - Column length validation has been completed.
28-Mar-22 16:33:03 - INFO - Missing values validation has been started.
28-Mar-22 16:33:03 - INFO - Raw data validation is completed.
28-Mar-22 16:33:03 - INFO - Creating training database and tables on the basis of given schema
28-Mar-22 16:33:15 - INFO - Opened websites keyspace successfully.
28-Mar-22 16:33:15 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
28-Mar-22 16:33:17 - INFO - Closed websites keyspace successfully.
28-Mar-22 16:33:17 - INFO - Table creation is completed.
28-Mar-22 16:33:17 - INFO - Insertion of data into the database is started.
28-Mar-22 16:33:24 - INFO - Opened websites keyspace successfully.
28-Mar-22 16:35:02 - INFO - The file has been inserted: no need to reinsert.
28-Mar-22 16:35:02 - INFO - Insertion into the database is completed.
28-Mar-22 16:35:02 - INFO - Deleting good data folder
28-Mar-22 16:35:02 - INFO - GoodRaw directory deleted successfully.
28-Mar-22 16:35:02 - INFO - Good_Data folder is deleted
28-Mar-22 16:35:02 - INFO - Moving bad files to archive and deleting Bad_Data folder
28-Mar-22 16:35:02 - INFO - Bad files moved to archive
28-Mar-22 16:35:02 - INFO - Bad Raw Data Folder Deleted successfully.
28-Mar-22 16:35:02 - INFO - Bad files are moved to archive. Bad folder is deleted.
28-Mar-22 16:35:02 - INFO - Validation operation is completed.
28-Mar-22 16:35:02 - INFO - Extracting csv file from the database
28-Mar-22 16:35:12 - INFO - Opened websites keyspace successfully.
28-Mar-22 16:36:48 - INFO - The csv file is exported successfully.
28-Mar-22 16:36:49 - INFO - Start of model training
28-Mar-22 16:36:49 - INFO - Entered the get_data method of the DataGetter class
28-Mar-22 16:36:50 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
28-Mar-22 16:36:50 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 16:36:51 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 16:36:51 - INFO - Entered the is_null_present method of the Preprocessor class
28-Mar-22 16:36:51 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
28-Mar-22 16:36:52 - INFO - Entered the replace_missing_values method of the Preprocessor class
28-Mar-22 16:36:56 - INFO - The -1 values have been replaced with -999.
28-Mar-22 16:36:56 - INFO - Entered the separate_label_feature method of the Preprocessor class
28-Mar-22 16:36:56 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
28-Mar-22 16:36:56 - INFO - Entered the elbow_plot method of the KMeansClustering class
28-Mar-22 16:37:46 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
28-Mar-22 16:37:46 - INFO - Entered the create_clusters method of the KMeansClustering class
28-Mar-22 16:37:47 - INFO - Entered the save_model method of the FileOperation class
28-Mar-22 16:37:48 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
28-Mar-22 16:37:48 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
28-Mar-22 16:37:49 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 16:37:50 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 16:37:50 - INFO - Entered the standardize_data method of the Preprocessor class
28-Mar-22 16:37:51 - INFO - The data has been standardized
28-Mar-22 16:37:52 - INFO - Entered the get_best_model method of the ModelFinder class
28-Mar-22 16:37:52 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 16:42:20 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 8, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 16:42:20 - INFO - ROC-AUC for XGBoost: 0.7527903317623194
28-Mar-22 16:42:20 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 16:54:37 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 16:54:40 - INFO - ROC-AUC for SVM: 0.5
28-Mar-22 16:54:40 - INFO - Entered the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 16:55:17 - INFO - LogisticRegression best params: {'C': 0.01, 'random_state': 0}             Exited the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 16:55:17 - INFO - ROC-AUC for LogisticRegression: 0.5067032979734787
28-Mar-22 16:55:17 - INFO - ROC-AUC for MixedNaiveBayes: 0.5011760932321181
28-Mar-22 16:55:17 - INFO - Entered the get_best_params_for_rf method of the ModelFinder class
28-Mar-22 16:57:05 - ERROR - Exception occured in get_best_params_for_rf method of the ModelFinder class.             Exception message: Input contains NaN, infinity or a value too large for dtype('float64').
28-Mar-22 16:57:05 - INFO - RandomForest training failed. Exited the get_best_params_for_rf method of             the ModelFinder class
28-Mar-22 16:57:05 - ERROR - Exception occurred in get_best_model method of the ModelFinder class.             Exception message: 
28-Mar-22 16:57:05 - INFO - Model selection Failed. Exited the get_best_model method of the Model_Finder class
28-Mar-22 16:57:05 - ERROR - Unsuccessful end of training
Traceback (most recent call last):
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 179, in maximize
    x_probe = next(self._queue)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 25, in __next__
    raise StopIteration("Queue is empty, no more objects to retrieve.")
StopIteration: Queue is empty, no more objects to retrieve.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 167, in get_best_params_for_rf
    rf_bo_result.maximize(init_points=45, n_iter=20)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 182, in maximize
    x_probe = self.suggest(util)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 128, in suggest
    self._gp.fit(self._space.params, self._space.target)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\gaussian_process\_gpr.py", line 230, in fit
    dtype=dtype,
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\base.py", line 581, in _validate_data
    X, y = check_X_y(X, y, **check_params)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 979, in check_X_y
    y = _check_y(y, multi_output=multi_output, y_numeric=y_numeric)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 990, in _check_y
    y, accept_sparse="csr", force_all_finite=True, ensure_2d=False, dtype=None
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 800, in check_array
    _assert_all_finite(array, allow_nan=force_all_finite == "allow-nan")
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 116, in _assert_all_finite
    type_err, msg_dtype if msg_dtype is not None else X.dtype
ValueError: Input contains NaN, infinity or a value too large for dtype('float64').

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 426, in get_best_model
    self.rf = self.get_best_params_for_rf(train_x, train_y)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 188, in get_best_params_for_rf
    raise Exception()
Exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_model.py", line 142, in training_model
    best_model_name, best_model = model_finder.get_best_model(X_train, y_train, X_test,  y_test)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 486, in get_best_model
    raise Exception()
Exception
28-Mar-22 17:13:16 - INFO - Start of validation on files for training
28-Mar-22 17:13:16 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
28-Mar-22 17:13:17 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
28-Mar-22 17:13:17 - INFO - Column length validation has been started.
28-Mar-22 17:13:17 - INFO - Column length validation has been completed.
28-Mar-22 17:13:17 - INFO - Missing values validation has been started.
28-Mar-22 17:13:17 - INFO - Raw data validation is completed.
28-Mar-22 17:13:17 - INFO - Creating training database and tables on the basis of given schema
28-Mar-22 17:13:31 - INFO - Opened websites keyspace successfully.
28-Mar-22 17:13:32 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
28-Mar-22 17:13:32 - INFO - Closed websites keyspace successfully.
28-Mar-22 17:13:32 - INFO - Table creation is completed.
28-Mar-22 17:13:32 - INFO - Insertion of data into the database is started.
28-Mar-22 17:13:41 - INFO - Opened websites keyspace successfully.
28-Mar-22 17:16:27 - INFO - The file has been inserted: no need to reinsert.
28-Mar-22 17:16:27 - INFO - Insertion into the database is completed.
28-Mar-22 17:16:27 - INFO - Deleting good data folder
28-Mar-22 17:16:27 - INFO - GoodRaw directory deleted successfully.
28-Mar-22 17:16:27 - INFO - Good_Data folder is deleted
28-Mar-22 17:16:27 - INFO - Moving bad files to archive and deleting Bad_Data folder
28-Mar-22 17:16:27 - INFO - Bad files moved to archive
28-Mar-22 17:16:27 - INFO - Bad Raw Data Folder Deleted successfully.
28-Mar-22 17:16:27 - INFO - Bad files are moved to archive. Bad folder is deleted.
28-Mar-22 17:16:27 - INFO - Validation operation is completed.
28-Mar-22 17:16:27 - INFO - Extracting csv file from the database
28-Mar-22 17:16:40 - INFO - Opened websites keyspace successfully.
28-Mar-22 17:19:31 - INFO - The csv file is exported successfully.
28-Mar-22 17:19:31 - INFO - Start of model training
28-Mar-22 17:19:31 - INFO - Entered the get_data method of the DataGetter class
28-Mar-22 17:19:32 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
28-Mar-22 17:19:32 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 17:19:33 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 17:19:33 - INFO - Entered the is_null_present method of the Preprocessor class
28-Mar-22 17:19:33 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
28-Mar-22 17:19:33 - INFO - Entered the replace_missing_values method of the Preprocessor class
28-Mar-22 17:19:37 - INFO - The -1 values have been replaced with -999.
28-Mar-22 17:19:37 - INFO - Entered the separate_label_feature method of the Preprocessor class
28-Mar-22 17:19:38 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
28-Mar-22 17:19:38 - INFO - Entered the elbow_plot method of the KMeansClustering class
28-Mar-22 17:20:17 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
28-Mar-22 17:20:17 - INFO - Entered the create_clusters method of the KMeansClustering class
28-Mar-22 17:20:19 - INFO - Entered the save_model method of the FileOperation class
28-Mar-22 17:20:19 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
28-Mar-22 17:20:19 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
28-Mar-22 17:20:19 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 17:20:20 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 17:20:20 - INFO - Entered the standardize_data method of the Preprocessor class
28-Mar-22 17:20:22 - INFO - The data has been standardized
28-Mar-22 17:20:22 - INFO - Entered the get_best_model method of the ModelFinder class
28-Mar-22 17:20:22 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 17:24:49 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 8, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 17:24:49 - INFO - ROC-AUC for XGBoost: 0.7527903317623194
28-Mar-22 17:24:49 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 17:37:28 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 17:37:31 - INFO - ROC-AUC for SVM: 0.5
28-Mar-22 17:37:31 - INFO - Entered the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 17:38:08 - INFO - LogisticRegression best params: {'C': 0.01, 'random_state': 0}             Exited the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 17:38:08 - INFO - ROC-AUC for LogisticRegression: 0.5067032979734787
28-Mar-22 17:38:08 - INFO - ROC-AUC for MixedNaiveBayes: 0.5011760932321181
28-Mar-22 17:38:08 - INFO - Entered the get_best_params_for_rf method of the ModelFinder class
28-Mar-22 17:39:26 - ERROR - Exception occured in get_best_params_for_rf method of the ModelFinder class.             Exception message: Input contains NaN, infinity or a value too large for dtype('float64').
28-Mar-22 17:39:26 - INFO - RandomForest training failed. Exited the get_best_params_for_rf method of             the ModelFinder class
28-Mar-22 17:39:26 - ERROR - Exception occurred in get_best_model method of the ModelFinder class.             Exception message: 
28-Mar-22 17:39:26 - INFO - Model selection Failed. Exited the get_best_model method of the Model_Finder class
28-Mar-22 17:39:26 - ERROR - Unsuccessful end of training
Traceback (most recent call last):
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 179, in maximize
    x_probe = next(self._queue)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 25, in __next__
    raise StopIteration("Queue is empty, no more objects to retrieve.")
StopIteration: Queue is empty, no more objects to retrieve.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 167, in get_best_params_for_rf
    rf_bo_result.maximize(init_points=45, n_iter=100)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 182, in maximize
    x_probe = self.suggest(util)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 128, in suggest
    self._gp.fit(self._space.params, self._space.target)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\gaussian_process\_gpr.py", line 230, in fit
    dtype=dtype,
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\base.py", line 581, in _validate_data
    X, y = check_X_y(X, y, **check_params)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 979, in check_X_y
    y = _check_y(y, multi_output=multi_output, y_numeric=y_numeric)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 990, in _check_y
    y, accept_sparse="csr", force_all_finite=True, ensure_2d=False, dtype=None
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 800, in check_array
    _assert_all_finite(array, allow_nan=force_all_finite == "allow-nan")
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 116, in _assert_all_finite
    type_err, msg_dtype if msg_dtype is not None else X.dtype
ValueError: Input contains NaN, infinity or a value too large for dtype('float64').

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 426, in get_best_model
    self.rf = self.get_best_params_for_rf(train_x, train_y)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 188, in get_best_params_for_rf
    raise Exception()
Exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_model.py", line 142, in training_model
    best_model_name, best_model = model_finder.get_best_model(X_train, y_train, X_test,  y_test)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 486, in get_best_model
    raise Exception()
Exception
28-Mar-22 18:14:20 - INFO - Start of validation on files for training
28-Mar-22 18:14:29 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
28-Mar-22 18:14:57 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
28-Mar-22 18:14:57 - INFO - Column length validation has been started.
28-Mar-22 18:14:57 - INFO - Column length validation has been completed.
28-Mar-22 18:14:57 - INFO - Missing values validation has been started.
28-Mar-22 18:14:57 - INFO - Raw data validation is completed.
28-Mar-22 18:14:57 - INFO - Creating training database and tables on the basis of given schema
28-Mar-22 18:15:38 - INFO - Opened websites keyspace successfully.
28-Mar-22 18:15:40 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
28-Mar-22 18:15:42 - INFO - Closed websites keyspace successfully.
28-Mar-22 18:15:42 - INFO - Table creation is completed.
28-Mar-22 18:15:42 - INFO - Insertion of data into the database is started.
28-Mar-22 18:15:54 - INFO - Opened websites keyspace successfully.
28-Mar-22 18:18:27 - INFO - The file has been inserted: no need to reinsert.
28-Mar-22 18:18:35 - INFO - Insertion into the database is completed.
28-Mar-22 18:18:35 - INFO - Deleting good data folder
28-Mar-22 18:18:38 - INFO - GoodRaw directory deleted successfully.
28-Mar-22 18:18:38 - INFO - Good_Data folder is deleted
28-Mar-22 18:18:38 - INFO - Moving bad files to archive and deleting Bad_Data folder
28-Mar-22 18:18:39 - INFO - Bad files moved to archive
28-Mar-22 18:18:39 - INFO - Bad Raw Data Folder Deleted successfully.
28-Mar-22 18:18:39 - INFO - Bad files are moved to archive. Bad folder is deleted.
28-Mar-22 18:18:39 - INFO - Validation operation is completed.
28-Mar-22 18:18:39 - INFO - Extracting csv file from the database
28-Mar-22 18:19:32 - INFO - Opened websites keyspace successfully.
28-Mar-22 18:23:08 - INFO - The csv file is exported successfully.
28-Mar-22 18:23:09 - INFO - Start of model training
28-Mar-22 18:23:10 - INFO - Entered the get_data method of the DataGetter class
28-Mar-22 18:23:17 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
28-Mar-22 18:23:17 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 18:23:20 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 18:23:20 - INFO - Entered the is_null_present method of the Preprocessor class
28-Mar-22 18:23:20 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
28-Mar-22 18:23:21 - INFO - Entered the replace_missing_values method of the Preprocessor class
28-Mar-22 18:23:26 - INFO - The -1 values have been replaced with -999.
28-Mar-22 18:23:26 - INFO - Entered the separate_label_feature method of the Preprocessor class
28-Mar-22 18:23:26 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
28-Mar-22 18:23:27 - INFO - Entered the elbow_plot method of the KMeansClustering class
28-Mar-22 18:25:34 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
28-Mar-22 18:25:34 - INFO - Entered the create_clusters method of the KMeansClustering class
28-Mar-22 18:25:35 - INFO - Entered the save_model method of the FileOperation class
28-Mar-22 18:25:40 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
28-Mar-22 18:25:42 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
28-Mar-22 18:25:46 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 18:25:46 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
28-Mar-22 18:25:46 - INFO - Entered the standardize_data method of the Preprocessor class
28-Mar-22 18:25:48 - INFO - The data has been standardized
28-Mar-22 18:25:49 - INFO - Entered the get_best_model method of the ModelFinder class
28-Mar-22 18:25:49 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 18:49:31 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 8, 'n_estimators': 130}             Exited the get_best_params_for_xgboost method of the ModelFinder class
28-Mar-22 18:49:34 - INFO - ROC-AUC for XGBoost: 0.7527903317623194
28-Mar-22 18:49:34 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 19:01:58 - INFO - SVM best params: {'C': 1.0, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
28-Mar-22 19:02:01 - INFO - ROC-AUC for SVM: 0.5
28-Mar-22 19:02:01 - INFO - Entered the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 19:02:40 - INFO - LogisticRegression best params: {'C': 0.01, 'random_state': 0}             Exited the get_best_params_for_logreg method of the ModelFinder class
28-Mar-22 19:02:40 - INFO - ROC-AUC for LogisticRegression: 0.5067032979734787
28-Mar-22 19:02:41 - INFO - ROC-AUC for MixedNaiveBayes: 0.5011760932321181
28-Mar-22 19:02:41 - INFO - Entered the get_best_params_for_rf method of the ModelFinder class
28-Mar-22 19:04:26 - ERROR - Exception occured in get_best_params_for_rf method of the ModelFinder class.             Exception message: Input contains NaN, infinity or a value too large for dtype('float64').
28-Mar-22 19:04:26 - INFO - RandomForest training failed. Exited the get_best_params_for_rf method of             the ModelFinder class
28-Mar-22 19:04:26 - ERROR - Exception occurred in get_best_model method of the ModelFinder class.             Exception message: 
28-Mar-22 19:04:26 - INFO - Model selection Failed. Exited the get_best_model method of the Model_Finder class
28-Mar-22 19:04:26 - ERROR - Unsuccessful end of training
Traceback (most recent call last):
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 179, in maximize
    x_probe = next(self._queue)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 25, in __next__
    raise StopIteration("Queue is empty, no more objects to retrieve.")
StopIteration: Queue is empty, no more objects to retrieve.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 167, in get_best_params_for_rf
    rf_bo_result.maximize(init_points=5, n_iter=100)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 182, in maximize
    x_probe = self.suggest(util)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\bayes_opt\bayesian_optimization.py", line 128, in suggest
    self._gp.fit(self._space.params, self._space.target)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\gaussian_process\_gpr.py", line 230, in fit
    dtype=dtype,
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\base.py", line 581, in _validate_data
    X, y = check_X_y(X, y, **check_params)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 979, in check_X_y
    y = _check_y(y, multi_output=multi_output, y_numeric=y_numeric)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 990, in _check_y
    y, accept_sparse="csr", force_all_finite=True, ensure_2d=False, dtype=None
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 800, in check_array
    _assert_all_finite(array, allow_nan=force_all_finite == "allow-nan")
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\validation.py", line 116, in _assert_all_finite
    type_err, msg_dtype if msg_dtype is not None else X.dtype
ValueError: Input contains NaN, infinity or a value too large for dtype('float64').

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 426, in get_best_model
    self.rf = self.get_best_params_for_rf(train_x, train_y)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 188, in get_best_params_for_rf
    raise Exception()
Exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_model.py", line 142, in training_model
    best_model_name, best_model = model_finder.get_best_model(X_train, y_train, X_test,  y_test)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 486, in get_best_model
    raise Exception()
Exception
29-Mar-22 10:44:31 - INFO - Start of validation on files for training
29-Mar-22 10:44:33 - INFO - LengthOfDateStampInFile:: 8 
                          LengthOfTimeStampInFile:: 6 
                          NumberofColumns:: 112
29-Mar-22 10:45:07 - INFO - Invalid File Name. phishing_09032022_ 082200.csv moved to Bad_Raw folder.
29-Mar-22 10:45:07 - INFO - Column length validation has been started.
29-Mar-22 10:45:07 - INFO - Column length validation has been completed.
29-Mar-22 10:45:07 - INFO - Missing values validation has been started.
29-Mar-22 10:45:07 - INFO - Raw data validation is completed.
29-Mar-22 10:45:07 - INFO - Creating training database and tables on the basis of given schema
29-Mar-22 10:46:07 - INFO - Opened websites keyspace successfully.
29-Mar-22 10:46:08 - INFO - Tables are in the database: no need to recreate them.
Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_data_insertion\db_operations.py", line 85, in create_table_db
    session.execute(create_query)
  File "cassandra\cluster.py", line 2618, in cassandra.cluster.Session.execute
  File "cassandra\cluster.py", line 4894, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'websites.phishing1' already exists
29-Mar-22 10:46:10 - INFO - Closed websites keyspace successfully.
29-Mar-22 10:46:10 - INFO - Table creation is completed.
29-Mar-22 10:46:10 - INFO - Insertion of data into the database is started.
29-Mar-22 10:46:21 - INFO - Opened websites keyspace successfully.
29-Mar-22 10:49:08 - INFO - The file has been inserted: no need to reinsert.
29-Mar-22 10:49:16 - INFO - Insertion into the database is completed.
29-Mar-22 10:49:16 - INFO - Deleting good data folder
29-Mar-22 10:49:19 - INFO - GoodRaw directory deleted successfully.
29-Mar-22 10:49:19 - INFO - Good_Data folder is deleted
29-Mar-22 10:49:19 - INFO - Moving bad files to archive and deleting Bad_Data folder
29-Mar-22 10:49:25 - INFO - Bad files moved to archive
29-Mar-22 10:49:25 - INFO - Bad Raw Data Folder Deleted successfully.
29-Mar-22 10:49:25 - INFO - Bad files are moved to archive. Bad folder is deleted.
29-Mar-22 10:49:25 - INFO - Validation operation is completed.
29-Mar-22 10:49:25 - INFO - Extracting csv file from the database
29-Mar-22 10:50:15 - INFO - Opened websites keyspace successfully.
29-Mar-22 10:53:30 - INFO - The csv file is exported successfully.
29-Mar-22 10:53:31 - INFO - Start of model training
29-Mar-22 10:53:32 - INFO - Entered the get_data method of the DataGetter class
29-Mar-22 10:53:34 - INFO - Data Load Successful. Exited the get_data method of the DataGetter class
29-Mar-22 10:53:35 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
29-Mar-22 10:53:44 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
29-Mar-22 10:53:44 - INFO - Entered the is_null_present method of the Preprocessor class
29-Mar-22 10:53:44 - INFO - Finding missing values is a success.Data written to the null values file.             Exited the is_null_present method of the Preprocessor class
29-Mar-22 10:53:48 - INFO - Entered the replace_missing_values method of the Preprocessor class
29-Mar-22 10:53:56 - INFO - The -1 values have been replaced with -999.
29-Mar-22 10:53:56 - INFO - Entered the separate_label_feature method of the Preprocessor class
29-Mar-22 10:53:56 - INFO - Label separation successful. Exited the separate_label_feature method of             the Preprocessor class
29-Mar-22 10:53:59 - INFO - Entered the elbow_plot method of the KMeansClustering class
29-Mar-22 10:56:02 - INFO - The optimum number of clusters is 3.             Exited the elbow_plot method of the KMeansClustering class
29-Mar-22 10:56:02 - INFO - Entered the create_clusters method of the KMeansClustering class
29-Mar-22 10:56:05 - INFO - Entered the save_model method of the FileOperation class
29-Mar-22 10:56:06 - INFO - Model File KMeans saved. Exited the save_model method of the FileOperation class
29-Mar-22 10:56:06 - INFO - Successfully created 3 clusters.             Exited the create_clusters method of the KMeansClustering class
29-Mar-22 10:56:12 - INFO - Entered the get_columns_with_zero_std_deviation method of the Preprocessor class
29-Mar-22 10:56:14 - INFO - Column search for standard deviation of zero is successful.             Exited the get_columns_with_zero_std_deviation method of the Preprocessor class
29-Mar-22 10:56:35 - INFO - Entered the standardize_data method of the Preprocessor class
29-Mar-22 10:56:38 - INFO - The data has been standardized
29-Mar-22 10:56:42 - INFO - Entered the get_best_model method of the ModelFinder class
29-Mar-22 10:56:42 - INFO - Entered the get_best_params_for_xgboost method of the ModelFinder class
29-Mar-22 11:03:17 - INFO - XGBoost best params: {'criterion': 'gini', 'max_depth': 9, 'n_estimators': 100}             Exited the get_best_params_for_xgboost method of the ModelFinder class
29-Mar-22 11:03:17 - INFO - ROC-AUC for XGBoost: 0.7427086554104623
29-Mar-22 11:03:17 - INFO - Entered the get_best_params_for_svm method of the ModelFinder class
29-Mar-22 11:16:46 - INFO - SVM best params: {'C': 0.5, 'kernel': 'rbf', 'random_state': 0}             Exited the get_best_params_for_svm method of the ModelFinder class
29-Mar-22 11:16:49 - INFO - ROC-AUC for SVM: 0.5
29-Mar-22 11:16:49 - INFO - Entered the get_best_params_for_logreg method of the ModelFinder class
29-Mar-22 11:17:42 - INFO - LogisticRegression best params: {'C': 3.0, 'random_state': 0}             Exited the get_best_params_for_logreg method of the ModelFinder class
29-Mar-22 11:17:42 - INFO - ROC-AUC for LogisticRegression: 0.5198960587873237
29-Mar-22 11:17:43 - INFO - ROC-AUC for MixedNaiveBayes: 0.5011760932321181
29-Mar-22 11:17:43 - INFO - Entered the get_best_params_for_rf method of the ModelFinder class
29-Mar-22 11:42:32 - ERROR - Exception occured in get_best_params_for_rf method of the ModelFinder class.             Exception message: min_samples_split must be an integer greater than 1 or a float in (0.0, 1.0]; got the integer 1
29-Mar-22 11:42:32 - INFO - RandomForest training failed. Exited the get_best_params_for_rf method of             the ModelFinder class
29-Mar-22 11:42:32 - ERROR - Exception occurred in get_best_model method of the ModelFinder class.             Exception message: 
29-Mar-22 11:42:32 - INFO - Model selection Failed. Exited the get_best_model method of the Model_Finder class
29-Mar-22 11:42:32 - ERROR - Unsuccessful end of training
joblib.externals.loky.process_executor._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\externals\loky\process_executor.py", line 436, in _process_worker
    r = call_item()
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\externals\loky\process_executor.py", line 288, in __call__
    return self.fn(*self.args, **self.kwargs)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\_parallel_backends.py", line 595, in __call__
    return self.func(*args, **kwargs)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\parallel.py", line 263, in __call__
    for func, args, kwargs in self.items]
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\parallel.py", line 263, in <listcomp>
    for func, args, kwargs in self.items]
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\fixes.py", line 216, in __call__
    return self.function(*args, **kwargs)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\model_selection\_validation.py", line 680, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\ensemble\_forest.py", line 467, in fit
    for i, t in enumerate(trees)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\parallel.py", line 1043, in __call__
    if self.dispatch_one_batch(iterator):
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\parallel.py", line 861, in dispatch_one_batch
    self._dispatch(tasks)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\parallel.py", line 779, in _dispatch
    job = self._backend.apply_async(batch, callback=cb)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\_parallel_backends.py", line 208, in apply_async
    result = ImmediateResult(func)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\_parallel_backends.py", line 572, in __init__
    self.results = batch()
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\parallel.py", line 263, in __call__
    for func, args, kwargs in self.items]
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\parallel.py", line 263, in <listcomp>
    for func, args, kwargs in self.items]
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\utils\fixes.py", line 216, in __call__
    return self.function(*args, **kwargs)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\ensemble\_forest.py", line 185, in _parallel_build_trees
    tree.fit(X, y, sample_weight=curr_sample_weight, check_input=False)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\tree\_classes.py", line 942, in fit
    X_idx_sorted=X_idx_sorted,
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\tree\_classes.py", line 254, in fit
    % self.min_samples_split
ValueError: min_samples_split must be an integer greater than 1 or a float in (0.0, 1.0]; got the integer 1
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 172, in get_best_params_for_rf
    self.grid.fit(train_x, train_y)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\skopt\searchcv.py", line 466, in fit
    super().fit(X=X, y=y, groups=groups, **fit_params)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\model_selection\_search.py", line 891, in fit
    self._run_search(evaluate_candidates)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\skopt\searchcv.py", line 514, in _run_search
    evaluate_candidates, n_points=n_points_adjusted
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\skopt\searchcv.py", line 408, in _step
    all_results = evaluate_candidates(params_dict)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\sklearn\model_selection\_search.py", line 851, in evaluate_candidates
    enumerate(candidate_params), enumerate(cv.split(X, y, groups))
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\parallel.py", line 1056, in __call__
    self.retrieve()
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\parallel.py", line 935, in retrieve
    self._output.extend(job.get(timeout=self.timeout))
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\site-packages\joblib\_parallel_backends.py", line 542, in wrap_future_result
    return future.result(timeout=timeout)
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\concurrent\futures\_base.py", line 435, in result
    return self.__get_result()
  File "C:\Users\Vahe\anaconda3\envs\phishing\lib\concurrent\futures\_base.py", line 384, in __get_result
    raise self._exception
ValueError: min_samples_split must be an integer greater than 1 or a float in (0.0, 1.0]; got the integer 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 553, in get_best_model
    self.rf = self.get_best_params_for_rf(train_x, train_y)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 236, in get_best_params_for_rf
    raise Exception()
Exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Vahe\PycharmProjects\pythonProject\training_model.py", line 142, in training_model
    best_model_name, best_model = model_finder.get_best_model(X_train, y_train, X_test,  y_test)
  File "C:\Users\Vahe\PycharmProjects\pythonProject\best_model_finder\tuner.py", line 613, in get_best_model
    raise Exception()
Exception
